python/mujoco/rollout_test.py

```diff
@@ -677,13 +706,17 @@ def py_rollout(model, data, initial_state, control,
   control = ensure_3d(control)
   nroll = initial_state.shape[0]
   nstep = control.shape[1]
-  nstate = mujoco.mj_stateSize(model, mujoco.mjtState.mjSTATE_FULLPHYSICS)
+
+  if isinstance(model, mujoco.MjModel):
+    model = [model]*nroll
```

Comment:
I see what you're doing here, but I'm not sure I like it. Imagine the "regular" case of single-model, multiple data. Say that I want a really wide batch 10k rollouts of 1 step each. In this case this line, whether called here or by the user could get very expensive due to 10k copies of MjModel.

The solution is for the pybind module and unsafe_rollout_ to correctly handle the special case where the list is of length 1, rather than to always require nroll MjModels. Does that make sense?

Reply:
So, this doesn't have a big performance impact because this copies the reference not the model. For 10k models, it takes 0.031 milliseconds to construct this list. The rest of the rollout functions checks (prior to calling into the bindings) takes 1.5 milliseconds. Taking one step over 10k `humanoid.xml` models takes about 250 milliseconds single threaded.

I can still make it a special case if wanted, but it won't affect performance. At the least a comment is in order.

(also this review comment is on the test, but the actual rollout function in rollout.py uses the same list expansion).

Reply:
Ah that's good to know! Resolving.